pkg/controllers/management/node/controller.go

```diff
@@ -413,6 +414,10 @@ func (m *Lifecycle) deployAgent(nodeDir string, obj *v3.Node) error {
 		return err
 	}
 
+	cluster.Spec, err = secretmigrator.AssembleRKEConfigSpec(cluster, cluster.Spec, m.credLister)
```

Comment:
We shouldn't be modifying the cluster spec, especially if the cluster gets accidentally saved.

Reply:
That doesn't it won't be saved in the future. We shouldn't be modifying the spec of objects.

Reply:
Now we make a deep copy of the cluster and modified the coped one, so we can avoid the accidental save by other processes. 
Can you review the PR again?